### PR TITLE
Fixing performance regression introduced earlier

### PR DESCRIPTION
**Pull request by hkaiser**
@tianyi93 Could you please verify whether this fixes your performance issue?

### CODE REVIEW

**Comment by biddisco** [comment]
The diff here is quite large and mostly stuff moved around, with some changes for 128 bit atomic presence - can you summarize please what is the difference that causes a speed improvement - thanks.

**Comment by hkaiser** [comment]
> The diff here is quite large and mostly stuff moved around, with some changes for 128 bit atomic presence - can you summarize please what is the difference that causes a speed improvement - thanks.

This PR is very small. The gist is we changed everything to `lockfree::queue` (from `lockfree::deque`), which has caused the issue.

**Comment by hkaiser** [comment]
> > The diff here is quite large and mostly stuff moved around, with some changes for 128 bit atomic presence - can you summarize please what is the difference that causes a speed improvement - thanks.
>>
> This actual change is very small. The gist is we changed everything to `lockfree::queue` (from `lockfree::deque`), which has caused the issue. See here for the important patch: https://github.com/STEllAR-GROUP/hpx/pull/4111/files#diff-6dece8d84d34dc5dc3d0df6e67f26f7dR37-R69.
>
> Everything else are changes caused by clang-format.

Sorry for this. I went back and redid the patch separating the actual fix from the reformatting.

**Comment by tianyi93** [comment]
@hkaiser I did a small test showing that the performance issue has been fixed. I will do the whole benchmark run. Thanks!

**Comment by hkaiser** [comment]
>  I did a small test showing that the performance issue has been fixed.

@tianyi93 Thanks! I'll merge this as soon as the tests have cycled.

**Comment by msimberg** [comment]
Have I understood correctly that `boost::lockfree::deque` is actually faster than `boost::lockfree::queue` as a FIFO queue? I admit I did just assume `queue` would be better than `deque`.

**Comment by hkaiser** [comment]
> Have I understood correctly that boost::lockfree::deque is actually faster than boost::lockfree::queue as a FIFO queue? I admit I did just assume queue would be better than deque.

@msimberg yes, that seems to be the case. I was not expecting this either. We ran into the performance regression while testing hpxMP. There is mostly use `hpx::apply` to schedule threads which are then joined using `hpx::lcos::local::latch`. Both facilities a very lightweight, thus the effect of the used scheduler was very pronounced.
